StarVMC/StarGeometry/CreateGeometry.cxx

```diff
@@ -0,0 +1,75 @@
+#include <TGeoManager.h>
+
+#include "StBFChain/StBFChain.h"
+#include "StarVMC/StarAgmlLib/AgModule.h"
+#include "StarVMC/StarAgmlLib/StarTGeoStacker.h"
+#include "StarVMC/StarGeometry/StarGeo.h"
+
+extern StBFChain* chain;
+
+/**
+ * Adapted from StarDb/VmcGeometry/CreateGeometry.h
+ */
+TDataSet *CreateGeometry(const Char_t *name="y2011") {
+  TObjectSet *geom = 0;
+  if ( gGeoManager ) { 
+    cout << "AgML geometry:  Existing TGeoManager " << gGeoManager->GetName() 
+	 << " detected, ignoring request for " 
+	 << name << endl;
+    return geom;
+  }
+
+  // Cache geometry to a TFile.  Geometry will be restored from TFile on subsequent calls.
+  TString filename = "";  
+  if  (chain)  { filename = chain->GetFileOut(); if ( filename=="" ) filename = chain->GetFileIn();  }
+  else { filename = name;  }
+
+  // Strip out @ symbol
+  filename = filename.ReplaceAll("@",""); 
+  // Strip off the last extention in the filename
+  filename = filename( 0, filename.Last('.') );
+  // Append geom.root to the extentionless filename
+  filename+=".geom.root";
+
+  // Detect second call to the system
+  if ( AgModule::Find("HALL") ) {
+    if ( chain->GetOption("Sti")    ||
+	 chain->GetOption("StiCA")  ||
+	 chain->GetOption("StiVMC") ){
+      cout << "AgML geometry:  HALL exists.  Restore from cache file " 
+	   << filename.Data() << endl;
+      gGeoManager = 0;
+      assert(0);
+      TGeoManager::Import( filename );
```

Comment:
This is a new behaviour, right? This may be surprising to users who expect that chains only write output ROOT files.

Reply:
No, nothing new --- since at least 2013 if I read the blame correctly. As noted in the description, the code is just copied from StarDb/AgMLGeometry/CreateGeometry.h as is. (There was an attempt to clean up this header in #462 but that broke something else, so can revisit if this compiled version works but in another PR) The CreateGeometry.h header is(was) "interpreted" (via other macros) by the snippet I removed from StRoot/StBFChain/StBFChain.cxx

Reply:
Oh, I though this code was somehow equivalent to the one removed from StBFChain. But, actually, it just relies on the fact that for each XXX, the StarDb/AgMLGeometry/Geometry.XXX.C only performs `CreateGeometry("XXX")`.

Thanks for looking up the blame. If this is an existing behavior, that's good to know, that's not an issue of this PR.